packages/tpetra/core/test/CrsGraph/CrsGraph_UnitTests_Swap.cpp

```diff
@@ -0,0 +1,387 @@
+        TEST_EQUALITY(graph.getNodeMaxNumRowEntries(), STMAX);                                                \
+        GST STGMAX;                                                                                           \
+        Teuchos::reduceAll<int, GST>(*STCOMM, Teuchos::REDUCE_MAX, STMAX, Teuchos::outArg(STGMAX));           \
+        TEST_EQUALITY(graph.getGlobalMaxNumRowEntries(), STGMAX);                                             \
+    }
+
+
+TEUCHOS_STATIC_SETUP()
+{
+    Teuchos::CommandLineProcessor& clp = Teuchos::UnitTestRepository::getCLP();
+    clp.setOption("filedir", &filedir, "Directory of expected input files.");
+    clp.addOutputSetupOptions(true);
+    clp.setOption("test-mpi",
+                  "test-serial",
+                  &Tpetra::TestingUtilities::testMpi,
+                  "Test MPI (if available) or force test of serial.  In a serial build,"
+                  " this option is ignored and a serial comm is always used.");
+    clp.setOption("error-tol-slack", &errorTolSlack, "Slack off of machine epsilon used to check test results");
+}
+// todo: update options so that this test won't run in serial (it's not set up for that currently)
+
+
+
+
+// comm           : a communicator with exactly 2 processes in it.
+// edges          : [ (u,v), ... ]
+// row_owners     : [ (u, pid), ... ]
+// gbl_num_columns: Max # of columns in the matrix-representation of the graph.
+//                  This should be >= the highest value of v from all edges (u,v) in edges.
+//                  Note: u and v are 0-indexed, so if the highest v is 11, then this should be 12.
+template<class LO, class GO, class Node, class comm_t>
+Teuchos::RCP<Tpetra::CrsGraph<LO, GO, Node>>
+generate_crsgraph(Teuchos::RCP<comm_t>                      comm,
+                  const std::vector<std::pair<GO, GO>>&     gbl_edges,
+                  const std::vector<std::pair<GO, size_t>>& gbl_row_owners,
+                  const size_t                              gbl_num_columns,
+                  const bool                                do_fillComplete=true)
+{
+    using Teuchos::Comm;
+
+    typedef Tpetra::CrsGraph<LO, GO, Node> graph_t;
+    typedef Tpetra::Map<LO, GO, Node>      map_t;
+
+    typedef typename std::map<GO, size_t> map_rows_t;      // row_id, num_entries
+
+
+    typedef typename std::vector<GO>        vec_go_t;
+    typedef typename std::map<GO, vec_go_t> map_row_to_cols_t;
+
+    const size_t comm_rank = comm->getRank();
+
+    map_row_to_cols_t gbl_rows;
+    for(auto& e: gbl_edges)
+    {
+        if(gbl_rows.find(e.first) == gbl_rows.end())
+        {
+            gbl_rows[ e.first ] = vec_go_t();
+        }
+        gbl_rows[ e.first ].push_back(e.second);
+    }
+
+    map_rows_t gbl_row2pid;
+    for(auto& p: gbl_row_owners) { gbl_row2pid.insert(p); }
+
+    // Print out some debugging information on what's in the graph
+    #if 0
```

Comment:
Use `Tpetra::Details::Behavior::verbose()` to control verbose output.

Reply:
I"m unfamiliar with this, are example uses of this pretty straightforward to find if I just search for `Tpetra::Details::Behavior::verbose()` in the testing dir?  If it's something with more obscure uses, what would be the thing I should search for to find it?

Reply:
```c++
const bool verbose = Tpetra::Details::Behavior::verbose();
if (verbose) {
  print_some_output();
}
```

Reply:
done